java/ql/lib/ext/java.awt.model.yml

```diff
@@ -6,11 +6,15 @@ extensions:
       - ["java.awt", "Container", True, "add", "(Component)", "", "Argument[0]", "Argument[this].Element", "value", "manual"]
       - ["java.awt", "Container", True, "add", "(Component)", "", "Argument[0]", "ReturnValue", "value", "manual"]
       - ["java.awt", "Container", True, "add", "(Component,Object)", "", "Argument[0]", "Argument[this].Element", "value", "manual"]
-
   - addsTo:
       pack: codeql/java-all
       extensible: neutralModel
     data:
       # The below APIs have numeric flow and are currently being stored as neutral models.
       # These may be changed to summary models with kinds "value-numeric" and "taint-numeric" (or similar) in the future.
       - ["java.awt", "Insets", "Insets", "(int,int,int,int)", "summary", "manual"] # value-numeric
+  - addsTo:
+      pack: codeql/java-all
+      extensible: sinkModel
+    data:
+      - ["java.awt", "Desktop", True, "browse", "(URI)", "", "Argument[0]", "request-forgery", "ai-manual"]
```

Comment:
this might be a misclassification, I'm not sure whether this might actually be a url-redirection (or something new altogether)

Reply:
Hard to tell.

On one hand, `url-redirection` is all about redirecting the user's browser, which is a separate component from the vulnerable application (the webapp running in a server), and in this case you're making a redirect (i.e. forcing a request) in the same machine where the vulnerable application lives, which sounds more like `request-forgery`.

But on the other hand, the fact that it redirects a _browser_, and doesn't really perform a server-to-server request makes it sound more like `url-redirection`.

I think neither match 1:1 in a desktop application, but if I had to choose I think I'd be slightly inclined to label it as `url-redirection` instead.

```suggestion
      - ["java.awt", "Desktop", True, "browse", "(URI)", "", "Argument[0]", "url-redirection", "ai-manual"]
```

Also, I don't expect many Java AWT applications to be left out there, so this probably won't have much impact :)

Reply:
Thanks, I've applied your suggestion.

> Also, I don't expect many Java AWT applications to be left out there, so this probably won't have much impact :)

["Write once, run anywhere!"](https://en.wikipedia.org/wiki/Write_once,_run_anywhere) ;)